### PR TITLE
fix(sdk): some cli fixes

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
- Allows for tab-completion of paths
- Fixes `kfp create recurring-run` bug
- Makes output logging more informative and consistent.
- fixes a broken f-string
- renames test methods from camel case to snake case

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/cli/component.py [hunk 1]
```diff
@@ -362,9 +360,9 @@ def component():
     is_flag=True,
     default=True,
     help='Push the built image to its remote repository.')
-def build(components_directory: pathlib.Path, component_filepattern: str,
-          engine: str, kfp_package_path: Optional[pathlib.Path],
-          overwrite_dockerfile: bool, push_image: bool):
+def build(components_directory: str, component_filepattern: str, engine: str,
+          kfp_package_path: Optional[str], overwrite_dockerfile: bool,
+          push_image: bool):
```
Comment (chensun) [inline sdk/python/kfp/cli/component.py:365]: nit: (just a small tip, you don't necessarily need to make a change) if you add a comma after the last parameter, formatting would be more predictable--each parameter on their own line.

**Review by chensun — COMMENTED** [review]
Should the title be `fix(sdk): ...` or `chore(sdk): ...` instead? I don't have a strong opinion, up to you.

/lgtm

Thanks @connor-mccarthy !

**Comment by connor-mccarthy** [comment]
Thanks, @chensun! The title should be `fix(sdk):...` -- I've updated it.

File: sdk/python/kfp/cli/component.py (hunk 1 above)
Comment (connor-mccarthy) [reply sdk/python/kfp/cli/component.py:365]: Thanks, @chensun. That's a great tip -- I will keep that in mind for future PRs.

**Pull request merged**